codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordDecoder.java

```diff
@@ -90,6 +94,57 @@ protected DnsRecord decodeRecord(
             String name, DnsRecordType type, int dnsClass, long timeToLive,
             ByteBuf in, int offset, int length) throws Exception {
 
+        if (type == DnsRecordType.A) {
```

Comment:
Besides, we hard to convert `DnsRecordType` to a `Enum`, because it use a private constructor wrap value of the unknown type, but `Enum` need a constant value.

```java
    public static DnsRecordType valueOf(int intValue) {
        DnsRecordType result = BY_TYPE.get(intValue);
        if (result == null) {
            return new DnsRecordType(intValue);
        }
        return result;
    }

    private DnsRecordType(int intValue) {
        this(intValue, "UNKNOWN");
    }
```